neqo-transport/src/connection.rs

```diff
@@ -1311,10 +1316,18 @@ impl Connection {
 
             out_bytes.append(&mut packet);
 
-            if self.role == Role::Client && *space == PNSpace::Handshake {
+            if *space == PNSpace::Handshake && self.role == Role::Client {
                 // Client can send Handshake packets -> discard Initial keys and states
                 self.discard_keys(PNSpace::Initial);
             }
+
+            if *space == PNSpace::Handshake
+                && self.role == Role::Server
+                && self.state == State::Confirmed
+            {
+                // We could discard handshake keys in set_state, but we are waiting to send an ack.
+                self.discard_keys(PNSpace::Handshake);
```

Comment:
Hmm, it would be really nice to do this in set_state, but I understand the value of having the ACK sent, even if it is only once.  There aren't a lot of places where we have an opportunity to do this discard otherwise.  This is fine for the moment, but I want to think about this a little more.